### PR TITLE
Sync OpenFAST restart PR to current amr-wind 

### PR DESCRIPTION
**Pull request by ndevelder**
This is a sync to latest amr-wind of https://github.com/Exawind/amr-wind/pull/485

Planning to test asap, will post results.

### CODE REVIEW

**Comment by ndevelder** [comment]
After first round of testing, this is close, we just need to update input flags to be consistent. "sim_mode" and "openfast_sim_mode" both seem to affect whether a case gets initialized, restarted, or replayed. If you want to use this PR right now, recommend using both in your input file.

**Comment by ndevelder** [comment]
This PR adds restarts that work as follows:

Note: current setup creates openfast .chkp file for every amr-wind restart slice. 

1. Change your `time.start_time` to the restart time
2. Set `io.restart_file` to your amr-wind restart slice directory
3. Set `Actuator.T1.openfast_sim_mode = restart` for each turbine (i.e. T1 in this case)
4. Set `Actuator.T1.openfast_restart_file` as the openfast checkpoint file. Note that this must be a relative path from your amr-wind case root, and the ".chkp" must be removed from the filename
5. Set `Actuator.T1.openfast_start_time` to the restart time and double check that `Actuator.T1.openfast_stop_time` is ok

**Review by lawrenceccheung — APPROVED** [review]
Thanks @ndevelder, we definitely need this capability.

**Comment by ndevelder** [comment]
And big thanks to @gantech for making it happen in the first place!

File: amr-wind/wind_energy/actuator/turbine/fast/FastIface.cpp [hunk 1]
```diff
@@ -295,10 +301,55 @@ void FastIface::fast_replay_turbine(FastTurbine& fi)
...
+
+    // Determine the number of substeps for FAST per CFD timestep
+    fi.num_substeps = static_cast<int>(std::floor(fi.dt_cfd / fi.dt_fast));
+
+    AMREX_ALWAYS_ASSERT(fi.num_substeps > 0);
+    // Check that the time step sizes are consistent and FAST advances at an
+    // integral multiple of CFD timestep
+    double dt_err =
+        fi.dt_cfd / (static_cast<double>(fi.num_substeps) * fi.dt_fast) - 1.0;
+    if (dt_err > 1.0e-4) {
```
Comment (psakievich) [inline amr-wind/wind_energy/actuator/turbine/fast/FastIface.cpp:348]: `1e-4` seems kind of generous.  We use `1e-12` in nalu-wind see [here](https://github.com/Exawind/nalu-wind/blob/531b51826358b6f1fe306a5f71ade571e0addf36/src/aero/actuator/ActuatorBulkFAST.C#L69-L70).  IIRC this was due to precision issue @lawrenceccheung had early in his time at SNL where the ratio was close, but not quite exact and it led to really weird errors.

**Review by psakievich — APPROVED** [review]
LGTM. I would suggest changing your ratio tolerance, but I won't hold this up over it.

**Review by gantech — APPROVED** [review]
Thanks for getting this to the finish line.

**Pull request merged**